cmd/apps/helm_operator_app.go

```diff
@@ -0,0 +1,171 @@
+package apps
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallHelmOperator() *cobra.Command {
+	var operator = &cobra.Command{
+		Use:          "helm-operator",
+		Short:        "Install a the helm operator",
+		Long:         `Install a the helm operator`,
+		Example:      `arkade install helm-operator --namespace default`,
+		SilenceUsage: true,
+	}
+
+	operator.Flags().StringP("namespace", "n", "default", "The namespace used for installation")
+	operator.Flags().Bool("update-repo", true, "Update the helm repo")
+	operator.Flags().Bool("helm3", true, "Use helm3, if set to false uses helm2")
+
+	operator.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath := getDefaultKubeconfig()
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+
+		updateRepo, _ := operator.Flags().GetBool("update-repo")
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+		helm3, _ := command.Flags().GetBool("helm3")
+
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+		namespace, _ := command.Flags().GetString("namespace")
+		if namespace != "default" {
+			return fmt.Errorf(`to override the "default", install via tiller`)
+		}
+
+		clientArch, clientOS := env.GetClientArch()
+
+		fmt.Printf("Client: %s, %s\n", clientArch, clientOS)
+		log.Printf("User dir established as: %s\n", userPath)
+
+		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS, helm3)
+		if err != nil {
+			return err
+		}
+
+		err = addHelmRepo("fluxcd", "https://charts.fluxcd.io", helm3)
+		if err != nil {
+			return err
+		}
+
+		if updateRepo {
+			err = updateHelmRepos(helm3)
+			if err != nil {
+				return err
+			}
+		}
+
+		chartPath := path.Join(os.TempDir(), "charts")
+		err = fetchChart(chartPath, "fluxcd/helm-operator", helm3)
+
+		if err != nil {
+			return err
+		}
+
+		overrides := map[string]string{}
+		// the helm chart always installs the crds
+		//overrides["createCRD"] = "true"
+		overrides["helm.versions"] = "v3 "
+
+		arch := getNodeArchitecture()
+
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		// Only so long until we get arm support
+		if arch == "arm64" {
```

Comment:
I would rather we don't have any third-party or unsupported images, I don't know `onedr0p`, so I going to ask for this to be removed. Sorry about that.

Reply:
Should I exit with a error on arm plattforms?

Reply:
Please 👍 

Folks can use an unsupported version by passing `--set image.repository` if they really want to, so this isn't required to be hard-coded.